cli/bpmetadata/parser/state_parser.go

```diff
@@ -0,0 +1,90 @@
+package parser
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"google.golang.org/protobuf/types/known/structpb"
+
+	tfjson "github.com/hashicorp/terraform-json"
+	"github.com/zclconf/go-cty/cty"
+)
+
+func ParseOutputTypesFromState(stateData []byte) (map[string]*structpb.Value, error) {
+	// Unmarshal the state data into a map[string]interface{} first
+	var rawState map[string]interface{}
+	err := json.Unmarshal(stateData, &rawState)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal state data: %w", err)
+	}
+
+	// Check if "format_version" key exists
+	if _, ok := rawState["format_version"]; !ok {
+		// If not present, add it with a default value
+		rawState["format_version"] = "1.0"
+	}
+
+	// Create a "values" field with "outputs" inside
+	rawState["values"] = map[string]interface{}{
+		"outputs": rawState["outputs"],
+	}
+
+	// Remove the top-level "outputs" field
+	delete(rawState, "outputs")
+
+	// Now marshal the updated map back to JSON
+	updatedStateData, err := json.MarshalIndent(rawState, "", "  ")
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal updated state data: %w", err)
+	}
+
+	var state tfjson.State
+
+	// Unmarshal the updated JSON into tfjson.State
+	err = json.Unmarshal(updatedStateData, &state)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal state data: %w", err)
+	}
+
+	outputTypeMap := make(map[string]*structpb.Value)
+	for name, output := range state.Values.Outputs {
+		pbValue, err := convertOutputTypeToStructpb(output)
+		if err != nil {
+			return nil, fmt.Errorf("failed to convert output %q to structpb.Value: %w", name, err)
+		}
+		outputTypeMap[name] = pbValue
+	}
+
+	return outputTypeMap, nil
+}
+
+func convertOutputTypeToStructpb(output *tfjson.StateOutput) (*structpb.Value, error) {
+	// Handle nil values explicitly
+	if output.Value == nil {
+		return structpb.NewNullValue(), nil
+	}
+
+	// Handle cases where output.Type is NilType
+	if output.Type == cty.NilType {
```

Comment:
Since we are generating output type for the blueprint, having NilType for an output variable doesn't provide any value? Why do we have a output variable with NilType?

Reply:
I don't think this is a valid case either. Just added this for robustness of the code. If this is indeed not a case I can either remove this or not set the type in `bpInterface`. @bharathkkb WDYT?

Reply:
This is also not a valid case AFAIK but we can keep it here as a check.